Middleware/InfoMiddleware.cs

```diff
@@ -0,0 +1,126 @@
+        private readonly String _disabledMiddleware;
+
+        private IEnumerable<IChatServiceConnection> chatServiceConnections;
+
+        public InfoMiddleware(ILogger<InfoMiddleware> logger,
+            ConfigurationSection configuration,
+            IEnumerable<IChatServiceConnection> chatServiceConnections,
+            BotMetadataProvider botMetadataProvider)
+        {
+            var orderedMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => mw.Enabled && mw.ExecutionOrderNumber != 0);
+            var unorderedMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => mw.Enabled && mw.ExecutionOrderNumber == 0 && mw.Configured);
+            var unconfiguredMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => mw.Enabled && !mw.Configured);
+            var disabledMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => !mw.Enabled);
+
+            // order middleware by 
+            //   1. ordered enabled MW in execution order
+            //   2. enabled MW that was not explicitly ordered
+            //   3. potentially enabled, but unconfigured MW
+
+            IEnumerable<string> orderedMiddlewareNamesList = new List<string>();
+            orderedMiddlewareNamesList = orderedMiddlewareNamesList.Concat(
+                orderedMiddlewareList.OrderBy(mw => mw.ExecutionOrderNumber).Select(mw => mw.Name)
+            );
+            orderedMiddlewareNamesList = orderedMiddlewareNamesList.Concat(
+                unorderedMiddlewareList.Select(mw => mw.Name)
+            );
+            orderedMiddlewareNamesList = orderedMiddlewareNamesList.Concat(
+                unconfiguredMiddlewareList.Select(mw => mw.Name)
+            );
+
+            _orderedMiddleware = string.Join(" -> ", orderedMiddlewareNamesList);
+            _alphabeticallyOrderedMiddleware = string.Join(", ", botMetadataProvider.RunningMiddlewares.Select(mw => mw.Enabled ? mw.Name : mw.Name + " (d)").OrderBy(str => str));
+            _disabledMiddleware = string.Join(", ", disabledMiddlewareList.Select(mw => mw.Name));
+
+            this.chatServiceConnections = chatServiceConnections;
+        }
+
+        public MiddlewareResult ProcessMessage(ChatMessageModel message)
+        {
+            // Ignore messages from ourself
+            if (message.IsFromTheorem)
+            {
+                return MiddlewareResult.Continue;
+            }
+
+            // Ignore messages that don't mention us, unless they're private messages
+            if (!message.IsMentioningTheorem && !message.IsPrivateMessage)
+            {
+                return MiddlewareResult.Continue;
+            }
+
+            var serviceConnection = message.FromChatServiceConnection;
+
+            var printList = message.Body.Contains("ordered") ? _orderedMiddleware : 
+                                (message.Body.Contains("disabled") ? _disabledMiddleware :_alphabeticallyOrderedMiddleware);
+
+            serviceConnection
```

Comment:
Hmmm maybe we should make ProcessMessage async

Reply:
I agree, I'll leave it for a future change, though, I don't wanna touch even more stuff than I've already touched here if that's okay with you :grimacing: